### PR TITLE
feat(anr-rate): Expose ANR rates through session API

### LINKED ISSUE #42468: Expose ANR rate through the sessions API
We've added a new tag to the release health metrics dataset
abnormal_mechanism in https://github.com/getsentry/relay/pull/1665. We want to use those tags
to calculate ANR rate and foreground ANR rate and expose
them through the sessions api.

ANR rate is defined as the ratio of unique users that have
experienced a background or foreground ANR / total unique users

Foreground ANR rate is defined as the ratio of unique users that have
experienced a foreground ANR / total unique users

### PR DESCRIPTION
**Pull request by shruthilayaj**
We've added a new tag to the release health metrics dataset
`abnormal_mechanism` in https://github.com/getsentry/relay/pull/1665. In this PR, we are using those tags
to calculate ANR rate and foreground ANR rate.

ANR rate is defined as the ratio of unique users that have
experienced a background or foreground ANR / total unique users

Foreground ANR rate is defined as the ratio of unique users that have
experienced a foreground ANR / total unique users

Add new derived metrics `anr_rate()` and `foreground_anr_rate()`
queryable through the sessions API. Note these fields are only supported
on the metrics backend and not on the sessions backend.

resolves https://github.com/getsentry/sentry/issues/42468

### CODE REVIEW

**Review by iambriccardo — CHANGES_REQUESTED** [review]
Thanks for the contribution Shruthi! In order to make sure no problems come up, I would suggest to test the new operations also in `tests/sentry/snuba/metrics/test_metrics_layer/test_release_health.py` with multiple cases (if needed) so that we can make sure that any added operator is properly tested. What do you think?

File: src/sentry/release_health/metrics_sessions_v2.py [hunk 1]
```diff
@@ -364,6 +364,8 @@ class SimpleForwardingField(Field):
         "crash_rate(user)": SessionMRI.CRASH_USER_RATE,
         "crash_free_rate(session)": SessionMRI.CRASH_FREE_RATE,
         "crash_free_rate(user)": SessionMRI.CRASH_FREE_USER_RATE,
+        "anr_rate()": SessionMRI.ANR_RATE,
+        "foreground_anr_rate()": SessionMRI.FOREGROUND_ANR_RATE,
```
Comment (iambriccardo) [inline src/sentry/release_health/metrics_sessions_v2.py:368]: Is there any reason for not putting `user` between `()`?

**Comment by iambriccardo** [comment]
Besides the two aforementioned comments, the changes look good to me.

File: src/sentry/release_health/metrics_sessions_v2.py (hunk 1 above)
Comment (shruthilayaj) [reply src/sentry/release_health/metrics_sessions_v2.py:368]: No reason other than that ANR rate by definition only accounts for unique users and it's not really parametrizable to take in any other input other than users

Comment (iambriccardo) [reply src/sentry/release_health/metrics_sessions_v2.py:368]: Great! I thought so by looking at the derived metrics.

**Review by iambriccardo — APPROVED** [review]
LGTM

**Pull request merged**